### PR TITLE
Add initial support for invites / users / groups

### PR DESCRIPTION
**Pull request by mattm-fugue**
Adds the following commands to the CLI for new invites / users / groups API endpoints:

```
fugue create invite --email test@example.com --expires true|false --group-ids default-admin-group
fugue get invite 4b04a829-47ed-44e7-b0a0-4405b3cdec60
fugue get user 4b04a829-47ed-44e7-b0a0-4405b3cdec60
fugue list users
fugue list users --email test@example.com
fugue list invites
fugue list invites --email test@example.com
fugue create group --name test --policy fugue:EDITOR --environment-ids 4b04a829-47ed-44e7-b0a0-4405b3cdec60,7869f9cf-5f37-409f-8a69-30dec4340c22
fugue create group --name test --policy fugue:EDITOR --all-environments
fugue list groups
fugue update users_groups --user-ids 4b04a829-47ed-44e7-b0a0-4405b3cdec60 --group-ids 7869f9cf-5f37-409f-8a69-30dec4340c22
```

Notes:
- When a user accepts an invite, the invite is no longer accessible via `fugue get invite`

### CODE REVIEW

File: cmd/createGroup.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	"github.com/fugue/fugue-client/client/groups"
+	"github.com/fugue/fugue-client/client/environments"
+	"github.com/fugue/fugue-client/format"
+	"github.com/fugue/fugue-client/models"
+	"github.com/spf13/cobra"
+)
+
+type createGroupOptions struct {
+	Name                     string
+    EnvironmentIds			 []string
```
Comment (eric-luminal) [inline cmd/createGroup.go:16]: formatting?

**Comment by eric-luminal** [comment]
do we have any way of users being able to grab the list of available policies? or is that just going to be in the documentation?

File: cmd/createGroup.go (hunk 1 above)
Comment (mattm-fugue) [reply cmd/createGroup.go]: Thanks, VSCode setup got messed up, all fixed now.

**Comment by mattm-fugue** [comment]
> do we have any way of users being able to grab the list of available policies? or is that just going to be in the documentation?

Documentation is the current plan. If you provide an incorrect value, the API will return you a list:

```
fugue create group --name all-envs-readonly --policy fugue:INCORRECT --all-environments
error: [POST /groups][400] createGroupBadRequest  &{Code:400 Message:policy must be one of ['fugue:READONLY', 'fugue:AUDITOR', 'fugue:EDITOR', 'fugue:CONTRIBUTOR'] Type:InvalidParameterValue}
```

**Review by eric-luminal — APPROVED** [review]

**Pull request merged**